src/InputOutput/VTK/fieldwriter.jl

```diff
@@ -0,0 +1,126 @@
+import KernelAbstractions: CPU
+
+using MPI
+using Printf
+
+using ..BalanceLaws
+using ..MPIStateArrays
+using ..DGMethods: SpaceDiscretization
+using ..VariableTemplates
+
+"""
+    VTKFieldWriter(
+        name::String,
+        FT::DataType,
+        fields::Vector{<:Tuple{String, <:Function}};
+        path_prefix = ".",
+        number_sample_points = 0,
+    )
+
+Construct a callable type that computes the specified `fields` and
+writes them to a VTK file in `path_prefix`/. `number_sample_points`
+are passed to [`writevtk`](@ref).
+
+Intended for use in a callback passed to a running simulation; files
+are suffixed with a running number beginning with 0.
+
+# Example
+```julia
+    function pres_fun(atmos::AtmosModel, prognostic::Vars, auxiliary::Vars)
+        ts = recover_thermo_state(atmos, prognostic, auxiliary)
+        air_pressure(ts)
+    end
+    fwriter = VTKFieldWriter(solver_config.name, [("pressure", pres_fun)])
+    cbfw = GenericCallbacks.EveryXSimulationTime(60) do
+        fwriter(solver_config.dg, solver_config.Q)
+    end
+```
+"""
+mutable struct VTKFieldWriter
+    path_prefix::String
+    name::String
+    number_sample_points::Int
+    nfields::Int
+    field_names::Vector{String}
+    field_funs::Vector{<:Function}
+    vars_type::DataType
+    num::Int
+
+    function VTKFieldWriter(
+        name::String,
+        FT::DataType,
+        fields::Vector{<:Tuple{String, <:Function}};
+        path_prefix = ".",
+        number_sample_points = 0,
```

Comment:
Minor, but other places this is a keyword argument.

Reply:
It's a keyword argument here also, as is `path_prefix`. :)

Reply:
Oops. Missed the semicolon, thought it was a default value!